htdocs/core/lib/functions.lib.php

```diff
@@ -2987,6 +2987,9 @@ function dol_print_date($time, $format = '', $tzoutput = 'auto', $outputlangs =
 		$format = 'daytextshort';
 	}
 
+	// Make sure the format strings are loaded/updated (#20209)
+	$outputlangs->load("main");
```

Comment:
dol_print_date is a low level memory function called a lot of time.
We should not have instruction into it that make IO like loading file from disk.
The main language file should be loaded by the caller that has initialized the $outputlangs object.

Reply:
You are right. I just found a better solution and will make new PR.